Src/ILGPU.Algorithms/MatrixOperations/MaskedMatrixProcessor.cs

```diff
@@ -0,0 +1,112 @@
+
+        /// <summary>
+        /// The internal masked matrix multiplier which contains pre-compiled kernels.
+        /// </summary>
+        private readonly MaskedSparseMatrixMultiplier<T, TPredicate, TStride>
+            matrixMultiplier;
+
+        /// <summary>
+        /// Constructs a new masked processor.
+        /// </summary>
+        /// <param name="accelerator">The parent accelerator.</param>
+        /// <param name="maxNumConcurrentStreams">
+        /// The maximum number of concurrent streams to use (if any).
+        /// </param>
+        /// <param name="streamProvider">
+        /// A custom stream provider function to construct specialized streams.
+        /// </param>
+        public MaskedMatrixProcessor(
+            Accelerator accelerator,
+            int maxNumConcurrentStreams = 0,
+            Func<Accelerator, AcceleratorStream> streamProvider = null)
+            : base(accelerator, maxNumConcurrentStreams, streamProvider)
+        {
+            matrixMultiplier = accelerator.CreateSparseMatrixMultiplierMasked<
+                T,
+                TPredicate,
+                TStride,
+                TProcessor>();
+        }
+
+        #endregion
+
+        /// <summary>
+        /// Returns the current predicate to use (if any).
+        /// </summary>
+        public TPredicate? Predicate { get; set; }
+
+        #region Methods
+        
+        /// <summary>
+        /// Multiplies the given matrices using the currently assigned predicate.
+        /// </summary>
+        /// <param name="stream">The current accelerator stream to use.</param>
+        /// <param name="aView">The dense input matrix a of shape MxK.</param>
+        /// <param name="bView">The sparse matrix b of shape NxK.</param>
+        /// <param name="outView">A dense output matrix of shape of aView.</param>
+        public void Multiply(
+            AcceleratorStream stream,
+            ArrayView2D<T, TStride> aView,
+            SparseMatrixView<T, TStride> bView,
+            ArrayView2D<T, TStride> outView)
+        {
+            if (!Predicate.HasValue)
+                throw new InvalidOperationException();
+            matrixMultiplier(stream, Predicate.Value, aView, bView, outView);
+        }
+
+        /// <summary>
+        /// Multiplies the given matrices using the currently assigned predicate.
+        /// </summary>
+        /// <param name="stream">The current accelerator stream to use.</param>
+        /// <param name="aViews">The dense input matrices a of shape MxK.</param>
+        /// <param name="bViews">The sparse matrices b of shape NxK.</param>
+        /// <param name="outViews">Dense output matrices of shape of aViews.</param>
+        public void MultiplyBatched(
+            AcceleratorStream stream,
+            IReadOnlyList<ArrayView2D<T, TStride>> aViews,
+            IReadOnlyList<SparseMatrixView<T, TStride>> bViews,
+            IReadOnlyList<ArrayView2D<T, TStride>> outViews)
```

Comment:
Is this unused?  Also, it says it uses the predicate but I don't see where that comes in.

Reply:
Not quite sure what you mean... the `Predicate` is a property defined in line 66.

Reply:
OK. I see now. Thanks.